test/mod_test_cases_test.dart

```diff
@@ -0,0 +1,39 @@
+import 'dart:io';
+
+import 'package:path/path.dart' show dirname;
+import './test_case.dart';
+
+/// This script performs snapshot testing of the inputs in the testing directory
+/// against golden files if they exist, and creates the golden files otherwise.
+///
+/// Input directory should be in `test/test_cases`, while the golden files should
+/// be in `test/test_cases_golden`.
+///
+/// For more information on the expected input and output, refer to the README
+void main() async {
+  var scriptPath = getScriptPath();
+  var testDirectory = Directory('$scriptPath/test_cases');
+  var goldDirectory = Directory('$scriptPath/test_cases_golden');
```

Comment:
nit: may I propose that we create directories `test/testdata/input` and `test/testdata/output` ?

Using a folder called `testdata` is idiomatic from golang, I figured we would just steal it :D

Also, let's create a `test/testdata/README.md` which explains:
 * How to run the tests
 * How the test input files are formatted,
 * How the test output files are formatted,
 * How to add test cases